presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardPredicate.java

```diff
@@ -105,21 +125,42 @@ public static ShardPredicate create(TupleDomain<RaptorColumnHandle> tupleDomain)
             }
 
             if (handle.isShardUuid()) {
-                // TODO: support multiple shard UUIDs
-                if (domain.isSingleValue()) {
-                    Slice uuidText = checkType(entry.getValue().getSingleValue(), Slice.class, "value");
-                    Slice uuidBytes;
-                    try {
-                        uuidBytes = uuidStringToBytes(uuidText);
+                StringJoiner rangePredicate = new StringJoiner(" OR ").setEmptyValue("true");
+
+                for (Range range : domain.getValues().getRanges().getOrderedRanges()) {
+                    if (range.isSingleValue()) {
+                        toShardPredicate(range.getSingleValue(), rangePredicate, "=")
+                                .ifPresent(value -> {
+                                    types.add(jdbcType);
+                                    values.add(value);
+                                });
                     }
-                    catch (IllegalArgumentException e) {
-                        predicate.add("false");
-                        continue;
+                    else {
+                        Marker high = range.getHigh();
+                        if (!high.isUpperUnbounded()) {
+                            toShardPredicate(high.getValue(), rangePredicate, high.getBound() == EXACTLY ? "<=" : "<")
```

Comment:
java.util.UUID uses mostSigBits and leastSigBits to compare UUIDs but `$shard_uuid` is defined as VARCHAR in Presto (probably because there's no UUID type) so I thought that it would be better to treat them as VARCHAR. I can change it if you wish, of course.